components/net/bluetooth_thread.rs

```diff
@@ -55,6 +55,55 @@ impl BluetoothThreadFactory for IpcSender<BluetoothMethodMsg> {
     }
 }
 
+fn matches_filter(device: &BluetoothDevice, filter: &BluetoothScanfilter) -> bool {
+    if filter.is_empty_or_invalid_filter() {
+        return false;
+    }
+
+     if !filter.get_name().is_empty() {
+        if let Ok(device_name) = device.get_name() {
+            if !device_name.eq(&filter.get_name()) {
+                return false;
+            }
+        } else {
+            return false;
+        }
+    }
+
+    if !filter.get_name_prefix().is_empty() {
+        if let Ok(device_name) = device.get_name() {
+            if !device_name.starts_with(&*filter.get_name_prefix()) {
+                return false;
+            }
+        } else {
+            return false;
+        }
+    }
+
+    if !filter.get_services().is_empty() {
+        if let Ok(device_uuids) = device.get_uuids() {
```

Comment:
i think your comment (at least the code) point to [line 64](https://github.com/szeged/servo/pull/21/files#diff-438a3422f5ba8c39dabd8e4e18928c51R64)
how about `if devices.get_name().ok() != Some(filter_name)` instead?